File: cucumber-messages/java/src/main/java/io/cucumber/messages/IdGenerator.java
package io.cucumber.messages;

public interface IdGenerator {
    String newId();

    class Incrementing implements IdGenerator {
        private int next = 0;

        @Override
        public String newId() {
            return Integer.toString(next++);
        }
    }

    class UUID implements IdGenerator {
        @Override
        public String newId() {
            return java.util.UUID.randomUUID().toString();
        }
    }
}

File: gherkin/dart/lib/src/language/IdGenerator.dart
import 'package:uuid/uuid.dart';

abstract class IdGenerator
{
  String newId();

  static final IdGenerator incrementingGenerator = Incrementing();

  static final IdGenerator uuidGenerator = UUID();
}

class UUID implements IdGenerator
{
  final Uuid _uuid = Uuid();

  @override
  String newId() => _uuid.v1();
}

class Incrementing implements IdGenerator
{
  int _next = 0;

  @override
  String newId() {
      _next++;
      return _next.toString();
  }
}

